src/tracer/MOM_tracer_types.F90

```diff
@@ -0,0 +1,131 @@
+!> Pulls out types previously defined in MOM_tracer_registry
+!! To avoid circular dependencies
+module MOM_tracer_types
+
+implicit none ; private
+
+#include <MOM_memory.h>
+
+!> The tracer type
+type, public :: tracer_type
+
+  real, dimension(:,:,:), pointer :: t              => NULL() !< tracer concentration array [conc]
```

Comment:
Again, I think all of the pointers here in `tracer_type` should be made allocable arrays, unless there is a limitation that I am unaware of.

Reply:
This is showing up as new code because I moved it out of `MOM_tracer_registry.F90`, but `t` is currently a pointer on `dev/ncar` so I think changing it is beyond the scope of this PR:

https://github.com/NCAR/MOM6/blob/dev/ncar/src/tracer/MOM_tracer_registry.F90#L40